src/view/prsTreeDataProvider.ts

```diff
@@ -121,7 +120,8 @@ export class PullRequestsTreeDataProvider implements vscode.TreeDataProvider<Tre
 	public async updateQueries() {
 		this._queries = await this.isVSO()
 			? []
-			: vscode.workspace.getConfiguration(SETTINGS_NAMESPACE, this._prManager.repository.rootUri).get<IQueryInfo[]>(QUERIES_SETTING) || [];
+			// TODO: is there a better scope to use?
+			: vscode.workspace.getConfiguration(SETTINGS_NAMESPACE, this._prManager.folderManagers[0]?.repository.rootUri).get<IQueryInfo[]>(QUERIES_SETTING) || [];
```

Comment:
I'm wondering if we could push the `queries` logic into `WorkspaceFolderNode` so that each one can use its own rootUri

Reply:
I guess this still needs to be here for the single folder case though

Reply:
Moved queries into `WorkspaceFolderNode` and reduced the duplicate code too.